### PR TITLE
Fix for issue with building sqlite3

### PR DESCRIPTION
**Pull request by msaxena2**
WIP, not yet ready. Fixes issue that occurs while building sqlite3 with rv-match, dealing with type of assignment expression.

### CODE REVIEW

**Comment by dwightguth** [comment]
@chathhorn can you review this once @msaxena2 says it's ready, since I basically worked very closely with him on it?

**Comment by msaxena2** [comment]
@chathhorn @dwightguth Please review.

**Comment by dwightguth** [comment]
looks good, but I want @chathhorn to take a look also

**Comment by chathhorn** [comment]
So I think this needs to be done basically anytime you're performing an lvalue conversion (but it isn't going through the normal channels, i.e. commom/expr/eval.k). With that in mind, I think you're missing cases and you're also doing it in places where it isn't needed (i.e., stripping qualifiers on the right operand, which is already lvalue-converted).

File: semantics/language/execution/expr/assignment.k [hunk 1]
```diff
@@ -39,7 +39,7 @@ module C-EXPR-ASSIGNMENT
           ~> checkArrayStaticAssign(T', V)
           ~> checkRestrictAssign(getRestrictBlock(T), getRelevantTags(Loc, V))
           ~> write(Loc, V, T')
-          ~> transferProv(Loc, tv(V, T))
+          ~> transferProv(Loc, tv(V, stripQualifiers(T)))
           requires isRestrictType(T)
```
Comment (chathhorn) [inline semantics/language/execution/expr/assignment.k]: The changes above shouldn't be needed, I don't think, or else the strip should happen sooner (when the actual lvalue conversion occurs).

**Pull request merged**